### ISSUE TITLE
JSON Sprite To String 

### ISSUE THREAD

**Issue by skyelynwaddell** [YoYoGames/GameMaker-Feature-Requests#1479]
### Is your feature request related to a problem?

Hey So I find it extremely annoying that I have to save the String of the sprite to JSON and then convert it to an asset and then have to reference that instead.

### Describe the solution you'd like

When I save a sprite inside a JSON file it should automatically save it as a string of that sprite and convert it upon load to the asset i had originally said to save. Make it detect im saving a sprite and save its name instead of just being saved as a reference to that sprite.
Thank you.

**Comment by BackYard321** [comment]
I feel like just changing the way JSONS work this way could cause other problems, but yeah honestly some kind of separate JSON stringifying/parsing functions that account for asset names (including objects, sprites, sound, etc) would be nice!

**Comment by skyelynwaddell** [comment]
> I feel like just changing the way JSONS work this way could cause other problems, but yeah honestly some kind of separate JSON stringifying/parsing functions that account for asset names (including objects, sprites, sound, etc) would be nice! Esp now that they're refs, rather than numbers.

yes i wouldnt want to make a hassle for users who have already built their systems using the current methods

**Comment by BackYard321** [comment]
The way I see it, storing an asset name in a JSON is a) more reliable, since you don't need to worry about the index #'s changing on you (which has happened to me), and b) actually readable, so even if you experience problems, you know the name of the asset that's creating issues for you. Plus, until we get functions like `asset_get_name()` and `asset_get_ref()`, you ALSO need to check whether to use `object_get_name()`, `sprite_get_name()`, etc, so this could also take much better advantage of refs. So i do think this would be really helpful functionality. 

That said, idk the logistics of actually programming something that can look at a string in a JSON and convert it to an asset. Would it have to look at every string and see whether it corresponds to an asset name? Would the string need a prefix or something like `object_index : "ASSET oSolid"`? So yeah again, idk whether it's feasible, but it would still be really useful.

**Comment by skyelynwaddell** [comment]
> Feature added in 2024.2
Thank you for adding this feature!! Im very happy to have contributed an idea into the game engine 💙💜

**Comment by skyelynwaddell** [comment]
i would assume it works just as it should, without the hassle of having to save it as a string in the json, you can just type the asset name. Same with when loading the asset, you wont have to convert it to a real anymore, but wont change systems already built with the string conversion method.

**Comment by gnysek** [comment]
@BackYard321 but it shows on Team Workload: https://github.com/orgs/YoYoGames/projects/17/views/7

**Comment by BackYard321** [comment]
> @BackYard321 but it shows on Team Workload: https://github.com/orgs/YoYoGames/projects/17/views/18

It was added yesterday - my message was from two weeks ago.
